plugins/module_utils/vmware_rest_client.py

```diff
@@ -333,12 +333,38 @@ def get_folder_by_name(self, datacenter_name, folder_name):
         datacenter = self.get_datacenter_by_name(datacenter_name)
         if not datacenter:
             return None
-        filter_spec = Folder.FilterSpec(type=Folder.Type.VIRTUAL_MACHINE,
-                                        names=set([folder_name]),
-                                        datacenters=set([datacenter]))
-        folder_summaries = self.api_client.vcenter.Folder.list(filter_spec)
-        folder = folder_summaries[0].folder if len(folder_summaries) > 0 else None
-        return folder
+        if "/" in folder_name:
+            folder_name = folder_name.strip('/').split('/')
+            p_folder_obj = None
+            for part in folder_name:
+                if p_folder_obj is None:
+                    if part == datacenter_name or part == "":
+                        continue
+                    else:
+                        filter_spec = Folder.FilterSpec(type=Folder.Type.VIRTUAL_MACHINE,
+                                                        names=set([part]),
+                                                        datacenters=set([datacenter]))
```

Comment:
```suggestion
                    filter_spec = Folder.FilterSpec(type=Folder.Type.VIRTUAL_MACHINE,
                                                    names=set([part]),
                                                    datacenters=set([datacenter]))
```

Reply:
I'll test this edit tonight. But I don't quite get it. 
If part is empty or is the datacenter is should skip to the next loop without searching for it. In this edit you always search even if  it is empty or has the DC name. Could be missing something but I'll try it out. :)